source/includes/connection-snippets/x509-snippets/c-connection.c

```diff
@@ -0,0 +1,32 @@
+// begin x509 connection
+#include <mongoc/mongoc.h>
+#include <stdio.h>
+
+int
+main (int argc, char *argv[])
+{
+   mongoc_collection_t *collection;
+   mongoc_client_t *client;
+   int64_t count;
+
+   mongoc_init ();
+
+   client = mongoc_client_new ("mongodb+srv://<cluster-url>/test?authSource=$external&tlsCertificateKeyFile=/etc/certs/mongodb/client.pem&retryWrites=true&w=majority&authMechanism=MONGODB-X509");
+
+   collection = mongoc_client_get_collection (client, "testDB", "testCol");
+   bson_t filter = BSON_INITIALIZER;
```

Comment:
@kevinAlbs Should all declarations appear at the top of the scope (`main` function body in this case)? I know C99 doesn't require it, but most of the example code for libmongoc follows that practice. This would apply to the `bson_t filter` and any `bson_error_t error` that may be added.

Reply:
I did not mention it before because it did not seem necessary to change, but if we're going with C89 standard for all docs examples, then yes the declaration should be moved up.

Reply:
Fixed - moved all declarations to the top